src/library-authoring/data/apiHooks.ts

```diff
@@ -121,6 +123,17 @@ export const useRevertLibraryChanges = () => {
     mutationFn: revertLibraryChanges,
     onSettled: (_data, _error, libraryId) => {
       queryClient.invalidateQueries({ queryKey: libraryAuthoringQueryKeys.contentLibrary(libraryId) });
+      queryClient.invalidateQueries({
+        // Invalidate all content queries related to this library
+        predicate: /* istanbul ignore next */ (query: Query): boolean => {
+          // extraFilter contains library id
+          const extraFilter = query.queryKey[5];
+          if (!(Array.isArray(extraFilter) || typeof extraFilter === 'string')) {
+            return false;
+          }
+          return query.queryKey[0] === 'content_search' && extraFilter?.includes(`context_key = "${libraryId}"`);
```

Comment:
We should add a comment here that if we allow searching "all courses and libraries" in the future, then we'd have to invalidate all `["content_search", "results"]` queries and not just the ones for this library, because items from this library could be included in an "all courses and libraries" search. But for now we only allow searching individual libraries one at a time (or "all courses") so this is fine.

Reply:
@bradenmacdonald Done.